src/test/ui/consts/const-mut-refs/mutable_references_ice.rs

```diff
@@ -19,7 +19,7 @@ unsafe impl Sync for Meh {}
 
 // the following will never be ok!
 const MUH: Meh = Meh {
-    x: &UnsafeCell::new(42), //~ WARN: skipping const checks
```

Comment:
This warning disappearing is several kinds of bad. An ICEing unleash test should not ever compile without the unleash flag.

Reply:
Should we keep an ICEing version for the unleash tests?

Reply:
with unleash this should not be a hard error. This test should not require any modifications.

Reply:
did you forget to re-add the unleash flag?

Reply:
I misunderstood you an tough we wanted to remove them. Should I revert the changes for both tests?